File: Code/CH5 Code/VB/MovieList-vb/Movie.vb
Option Strict On

Public Class Movie

  Private mName As String
  Private mSalePrice As Decimal
  Private mRentalPrice As Decimal

  Public Sub New(ByVal name As String)
    mName = name
    mSalePrice = 12.95D
    mRentalPrice = 3.4D
  End Sub

  Public ReadOnly Property Name() As String
    Get
      Return mName
    End Get
  End Property

  Public ReadOnly Property SalePrice() As Decimal
    Get
      Return mSalePrice
    End Get
  End Property

  Public ReadOnly Property RentalPrice() As Decimal
    Get
      Return mRentalPrice
    End Get
  End Property


End Class

File: Code/CH5 Code/C#/MovieList-c/Movie.cs
using System;

namespace MovieList_c
{
  /// <summary>
  /// Summary description for Movie.
  /// </summary>
  public class Movie
  {
    private string  mName;
    private Decimal mSalePrice;
    private Decimal mRentalPrice;

    public Movie(string name)
    {
      mName  = name;
      mSalePrice = 12.95m;
      mRentalPrice = 3.40m;
    }

    public string  Name        { get { return mName; } }
    public Decimal SalePrice   { get { return mSalePrice; } }
    public Decimal RentalPrice { get { return mRentalPrice; } }
  }
}
